src/poly-op.h

```diff
@@ -0,0 +1,27 @@
+#ifndef VCTRS_POLY_OP
+#define VCTRS_POLY_OP
+
+#include "vctrs.h"
+
+typedef int (*poly_binary_int_fn_t)(const void* x, r_ssize i, const void* y, r_ssize j);
```

Comment:
Should we use the `_fn_ptr` suffix instead of `_fn_t`? This is what I use in rlang. I find it makes it slightly clearer what the type is about.